# CE49X Lab 2: Is Wave Energy Worth the Investment?
## A Financial Feasibility Comparison of Renewable Energy in Perth, Australia

**Instructor:** Dr. Eyuphan Koc  
**Department of Civil Engineering, Bogazici University**  
**Semester:** Spring 2026

        {
            "cell_type": "code",
            "execution_count": null,
            "metadata": {
                "language": "python"
            },
            "outputs": [],
            "source": [
                "# Sensitivity analysis for Wave Energy NPV\n",
                "import numpy as np\n",
                "import matplotlib.pyplot as plt\n",
                "\n",
                "# Vary CAPEX and compute NPV (in million USD) for the Wave Energy technology\n",
                "capex_range = np.linspace(500, 7000, 60)  # USD/kW\n",
                "npv_vals = []\n",
                "for c in capex_range:\n",
                "    npv = calculate_npv(c, technologies['Wave Energy']['opex'], technologies['Wave Energy']['cf'], technologies['Wave Energy']['lifetime'], capacity_kw, discount_rate)\n",
                "    npv_vals.append(npv / 1e6)  # convert to million USD\n",
                "\n",
                "# Estimate break-even CAPEX where NPV crosses zero (if within range)\n",
                "break_even_capex = None\n",
                "if np.sign(npv_vals[0]) != np.sign(npv_vals[-1]):\n",
                "    break_even_capex = np.interp(0, np.array(npv_vals), capex_range)\n",
                "\n",
                "# Vary capacity factor and compute NPV (in million USD)\n",
                "cf_range = np.linspace(0.02, 0.5, 60)\n",
                "npv_cf = []\n",
                "for cf in cf_range:\n",
                "    npv = calculate_npv(technologies['Wave Energy']['capex'], technologies['Wave Energy']['opex'], cf, technologies['Wave Energy']['lifetime'], capacity_kw, discount_rate)\n",
                "    npv_cf.append(npv / 1e6)\n",
                "\n",
                "break_even_cf = None\n",
                "if np.sign(npv_cf[0]) != np.sign(npv_cf[-1]):\n",
                "    break_even_cf = np.interp(0, np.array(npv_cf), cf_range)\n",
                "\n",
                "# Plot NPV vs CAPEX\n",
                "plt.figure(figsize=(8, 4))\n",
                "plt.plot(capex_range, npv_vals, label='NPV (million USD)')\n",
                "plt.axhline(0, color='k', linestyle='--')\n",
                "plt.title('Wave Energy NPV vs CAPEX (100 MW)')\n",
                "plt.xlabel('CAPEX (USD/kW)')\n",
                "plt.ylabel('NPV (million USD)')\n",
                "if break_even_capex is not None:\n",
                "    plt.scatter([break_even_capex], [0], color='red')\n",
                "    plt.annotate(f'BE CAPEX={break_even_capex:.0f} USD/kW', (break_even_capex, 0), xytext=(10, 10), textcoords='offset points')\n",
                "plt.grid(True)\n",
                "plt.tight_layout()\n",
                "plt.show()\n",
                "\n",
                "# Plot NPV vs Capacity Factor\n",
                "plt.figure(figsize=(8, 4))\n",
                "plt.plot(cf_range, npv_cf, label='NPV (million USD)')\n",
                "plt.axhline(0, color='k', linestyle='--')\n",
                "plt.title('Wave Energy NPV vs Capacity Factor (100 MW)')\n",
                "plt.xlabel('Capacity Factor')\n",
                "plt.ylabel('NPV (million USD)')\n",
                "if break_even_cf is not None:\n",
                "    plt.scatter([break_even_cf], [0], color='red')\n",
                "    plt.annotate(f'BE CF={break_even_cf:.3f}', (break_even_cf, 0), xytext=(10, 10), textcoords='offset points')\n",
                "plt.grid(True)\n",
                "plt.tight_layout()\n",
                "plt.show()\n",
                "\n",
                "print('Break-even CAPEX (NPV=0):', f'{break_even_capex:.0f} USD/kW' if break_even_capex is not None else 'Not within tested CAPEX range')\n",
                "print('Break-even Capacity Factor (NPV=0):', f'{break_even_cf:.3f}' if break_even_cf is not None else 'Not within tested CF range')\n"
            ]
        },
        {
            "cell_type": "markdown",
            "id": "#VSC-65fab963",
            "metadata": {
                "language": "markdown"
            },
            "source": [
                "## 5. Recommendation"
            ]
        },

## Deliverables

Your notebook must include the following:

### 1. Wave Energy Analysis (from the dataset)
- Load and explore the `WEC_Perth_49.csv` dataset
- Compute statistics on farm power output (mean, min, max, standard deviation)
- Estimate a **capacity factor** for wave energy in Perth based on the data (you'll need to assume a rated capacity per WEC — research and justify your choice)

### 2. Competing Technologies
- Choose **at least two** other renewable energy technologies to compare against wave energy (e.g., solar PV, onshore wind, offshore wind, tidal, biomass)
- Research and cite the following for **each** technology (including wave):
  - Capital cost (CAPEX) per kW installed
  - Annual operating cost (OPEX) per kW
  - Capacity factor specific to the Perth region
  - Expected project lifetime
- **Cite your sources.** Use data from reputable organizations (e.g., IRENA, IEA, CSIRO, NREL, Lazard).

### 3. Financial Comparison
- Calculate the **Levelized Cost of Energy (LCOE)** for each technology
- Calculate **at least one additional financial metric** of your choice (e.g., NPV, payback period, internal rate of return, cost per annual MWh)
- Choose an appropriate **discount rate** and justify it

### 4. Visualization
- Create **at least two plots** that clearly communicate your comparison
- Plots should be publication-quality: labeled axes, title, legend, grid

### 5. Recommendation
- Based on your analysis, write a short recommendation (1-2 paragraphs):
  - Which technology (or mix) should Perth invest in?
  - Under what conditions could wave energy become competitive?
  - What factors does your financial model **not** capture?

## Hints

- **LCOE formula:**

$$\text{LCOE} = \frac{\text{Total Discounted Costs}}{\text{Total Discounted Energy}} = \frac{\text{CAPEX} + \sum_{t=1}^{N} \frac{\text{OPEX}_t}{(1+r)^t}}{\sum_{t=1}^{N} \frac{E_t}{(1+r)^t}}$$

  where $r$ is the discount rate, $N$ is the project lifetime, and $E_t$ is annual energy production in MWh.

- **Annual energy production:** $E = \text{Capacity (kW)} \times \text{Capacity Factor} \times 8760 \text{ hours/year}$

- The dataset gives power in **Watts**. Be careful with unit conversions.

- Think about what the dataset's `Total_Power` column actually represents and how it relates to the rated capacity of a real WEC device.

## Grading

| Component | Weight |
|-----------|--------|
| Wave energy analysis (dataset exploration, capacity factor) | 20% |
| Research quality (cost data, sources, justification) | 25% |
| Financial calculations (LCOE + additional metric) | 25% |
| Visualizations (clarity, quality) | 15% |
| Recommendation (insight, completeness) | 15% |

## Submission

1. Complete your work in **this notebook** on your own fork of the course repository.
2. Make sure your notebook **runs top-to-bottom without errors** before submitting.
3. Commit and push your completed notebook to your fork.
4. We will grade directly from your fork — there is no separate upload. Make sure your latest work is pushed before the deadline.

---
## Your Work Starts Here

## 1. Wave Energy Analysis

In [14]:
import pandas as pd
import numpy as np
import seaborn as sns

import matplotlib.pyplot as plt

# Load the WEC Perth dataset
df_wec = pd.read_csv('WEC_Perth_49.csv')

# Display basic information about the dataset
print("Dataset shape:", df_wec.shape)
print("\nFirst few rows:")
print(df_wec.head())
print("\nColumn names:")
print(df_wec.columns.tolist())
print("\nData types:")
print(df_wec.dtypes)
print("\nBasic statistics:")
print(df_wec.describe())

Dataset shape: (36043, 149)

First few rows:
      X1    Y1      X2     Y2      X3     Y3      X4      Y4     X5   Y5  ...  \
0  600.0   0.0  546.16  37.50  489.79  74.88  432.47  112.05  650.0  0.0  ...   
1  593.0  12.0  546.16  37.50  489.79  74.88  432.47  112.05  644.0  8.0  ...   
2  593.0  12.0  546.16  37.50  489.79  74.88  432.47  112.05  644.0  8.0  ...   
3  593.0  12.0  546.16  37.50  489.79  74.88  432.47  112.05  644.0  8.0  ...   
4  200.0   0.0  146.17  37.53   89.76  74.93   32.40  112.18  400.0  0.0  ...   

    Power42   Power43    Power44   Power45    Power46    Power47   Power48  \
0  88867.92  98844.30  101283.59  98934.63  101624.58  100915.03  99625.68   
1  88896.55  98759.79  101346.07  98873.59  101629.01  100934.53  99606.13   
2  88919.83  98746.68  101346.15  98875.57  101618.32  100941.00  99611.35   
3  88855.14  98760.96  101338.59  98971.58  101632.28  100943.59  99589.25   
4  88005.30  98630.24  100432.73  98803.01  101064.48  100948.38  99028.87   


## 2. Competing Technologies

In [10]:
# Compute statistics on farm power output
mean_power = df_wec['Total_Power'].mean()
min_power = df_wec['Total_Power'].min()
max_power = df_wec['Total_Power'].max()
std_power = df_wec['Total_Power'].std()

print(f"Mean farm power: {mean_power:.2f} W")
print(f"Min farm power: {min_power:.2f} W")
print(f"Max farm power: {max_power:.2f} W")
print(f"Std farm power: {std_power:.2f} W")

# Assume rated power per WEC is 500 kW = 500,000 W
# Justification: Based on typical wave energy converters like the Pelamis device, which has a rated power of around 750 kW, but for Perth's conditions and the dataset, we assume 500 kW per device as a conservative estimate. Source: IRENA Wave Energy report.
rated_per_wec = 500000  # W
total_rated_power = 49 * rated_per_wec  # W
capacity_factor_wave = mean_power / total_rated_power

print(f"Assumed rated power per WEC: {rated_per_wec / 1000:.0f} kW")
print(f"Total rated power for farm: {total_rated_power / 1000000:.1f} MW")
print(f"Estimated capacity factor for wave energy in Perth: {capacity_factor_wave:.3f}")

Mean farm power: 3938246.46 W
Min farm power: 3388944.15 W
Max farm power: 4177658.97 W
Std farm power: 122617.15 W
Assumed rated power per WEC: 500 kW
Total rated power for farm: 24.5 MW
Estimated capacity factor for wave energy in Perth: 0.161


---

### Questions?

**Dr. Eyuphan Koc**  
eyuphan.koc@bogazici.edu.tr

## 3. Financial Comparison

In [9]:

# Define technologies data
# Sources: IRENA Renewable Power Generation Costs in 2022 for CAPEX and OPEX.
# Capacity factors: For Perth, Australia - Solar PV: CSIRO data ~22%, Onshore Wind: NREL ~35%, Wave: calculated from dataset.
# Lifetimes: IRENA estimates.
import pandas as pd
# Ensure capacity_factor_wave is defined
if 'capacity_factor_wave' not in globals():
    if 'df_wec' not in globals():
        df_wec = pd.read_csv('WEC_Perth_49.csv')
    mean_power = df_wec['Total_Power'].mean()
    rated_per_wec = 500000  # W
    total_rated_power = 49 * rated_per_wec  # W
    capacity_factor_wave = mean_power / total_rated_power

technologies = {
    'Wave Energy': {
        'capex': 7000,  # USD/kW
        'opex': 200,    # USD/kW/year
        'cf': capacity_factor_wave,
        'lifetime': 20,  # years
        'source': 'IRENA 2022 (emerging technologies), CF from dataset analysis'
    },
    'Solar PV': {
        'capex': 1200,  # USD/kW
        'opex': 20,     # USD/kW/year
        'cf': 0.22,
        'lifetime': 25,  # years
        'source': 'IRENA 2022, CF: CSIRO for Perth region'
    },
    'Onshore Wind': {
        'capex': 1500,  # USD/kW
        'opex': 40,     # USD/kW/year
        'cf': 0.35,
        'lifetime': 25,  # years
        'source': 'IRENA 2022, CF: NREL for Western Australia'
    }
}

# Create and display technology parameters table
tech_df = pd.DataFrame.from_dict(technologies, orient='index')
tech_df = tech_df[['capex', 'opex', 'cf', 'lifetime']]
tech_df.columns = ['CAPEX (USD/kW)', 'OPEX (USD/kW/year)', 'Capacity Factor', 'Lifetime (years)']

print("Technology Parameters:")
print(tech_df)
print("\nSources:")
for tech, data in technologies.items():
    print(f"{tech}: {data['source']}")
print()

Technology Parameters:
              CAPEX (USD/kW)  OPEX (USD/kW/year)  Capacity Factor  \
Wave Energy             7000                 200         0.160745   
Solar PV                1200                  20         0.220000   
Onshore Wind            1500                  40         0.350000   

              Lifetime (years)  
Wave Energy                 20  
Solar PV                    25  
Onshore Wind                25  

Sources:
Wave Energy: IRENA 2022 (emerging technologies), CF from dataset analysis
Solar PV: IRENA 2022, CF: CSIRO for Perth region
Onshore Wind: IRENA 2022, CF: NREL for Western Australia



## 4. Visualization

In [ ]:
import numpy as np
import pandas as pd

# Discount rate: 5% - typical for renewable energy projects in Australia, reflecting moderate risk and inflation. Source: Based on Australian Treasury guidelines.
discount_rate = 0.05

# Project capacity: 100 MW as per the requirement
capacity_mw = 100  # MW
capacity_kw = capacity_mw * 1000  # kW

def calculate_lcoe(capex, opex, cf, lifetime, capacity_kw, discount_rate):
    """
    Calculate Levelized Cost of Energy (LCOE) in USD/MWh
    """
    # Annual energy production in MWh
    annual_energy_mwh = capacity_kw * cf * 8760 / 1000
    
    # Annuity factor
    annuity = (1 - (1 + discount_rate) ** -lifetime) / discount_rate
    
    # Discounted costs
    discounted_capex = capex * capacity_kw
    discounted_opex = opex * capacity_kw * annuity
    
    # Discounted energy
    discounted_energy = annual_energy_mwh * annuity
    
    lcoe = (discounted_capex + discounted_opex) / discounted_energy
    return lcoe

def calculate_npv(capex, opex, cf, lifetime, capacity_kw, discount_rate, electricity_price=0.08):
    """
    Calculate Net Present Value (NPV) in USD
    Assuming electricity price of 0.08 USD/kWh (approximate Australian wholesale price)
    """
    npv = -capex * capacity_kw
    annual_energy_mwh = capacity_kw * cf * 8760 / 1000
    annual_revenue = annual_energy_mwh * electricity_price * 1000  # USD
    annual_cost = opex * capacity_kw
    
    for t in range(1, lifetime + 1):
        npv += (annual_revenue - annual_cost) / (1 + discount_rate) ** t
    
    return npv

# Calculate LCOE and NPV for each technology
results = {}
for tech, data in technologies.items():
    lcoe = calculate_lcoe(data['capex'], data['opex'], data['cf'], data['lifetime'], capacity_kw, discount_rate)
    npv = calculate_npv(data['capex'], data['opex'], data['cf'], data['lifetime'], capacity_kw, discount_rate)
    results[tech] = {'LCOE': lcoe, 'NPV': npv}

# Create and display results table
results_df = pd.DataFrame.from_dict(results, orient='index')
results_df.columns = ['LCOE (USD/MWh)', 'NPV (million USD)']
results_df['NPV (million USD)'] = results_df['NPV (million USD)'] / 1e6  # Convert to millions

print(f"Discount rate: {discount_rate * 100}%")
print(f"Project capacity: {capacity_mw} MW")
print(f"Assumed electricity price: $0.08/kWh")
print("\nFinancial Results:")
print(results_df)

Discount rate: 5.0%
Project capacity: 100 MW
Assumed electricity price: $0.08/kWh

Financial Results:
              LCOE (USD/MWh)  NPV (million USD)
Wave Energy       540.931118        -808.857503
Solar PV           54.557362          69.106911
Onshore Wind       47.758867         139.320494


## 5. Recommendation

Based on the financial analysis, Perth should prioritize investing in **Onshore Wind** for the 100 MW renewable energy capacity addition, as it offers the lowest LCOE at approximately $48/MWh compared to $55/MWh for solar PV and $541/MWh for wave energy. The NPV for onshore wind is also the highest at $139 million, indicating strong profitability. Solar PV follows as a close second with competitive costs.

Wave energy, while promising for its predictability and minimal visual impact, is currently not competitive due to high capital costs and lower capacity factor derived from the dataset. It could become viable if technological advancements reduce CAPEX by 80% or more, or if policy incentives like feed-in tariffs specifically target emerging technologies. Additionally, wave energy might be considered in a diversified portfolio to balance intermittency of solar and wind.

The model does not capture environmental benefits, grid integration costs, land use impacts, or potential future cost reductions through learning curves. It also assumes constant capacity factors and costs, without accounting for degradation over time or regional variations in resource availability.

In [10]:
# Plot 1: LCOE Comparison
tech_names = list(results.keys())
lcoe_values = [results[tech]['LCOE'] for tech in tech_names]

plt.figure(figsize=(8, 5))
plt.bar(tech_names, lcoe_values, color=['blue', 'green', 'red'])
plt.title('Levelized Cost of Energy (LCOE) Comparison', fontsize=14)
plt.ylabel('LCOE (USD/MWh)', fontsize=12)
plt.xlabel('Technology', fontsize=12)
plt.grid(axis='y', linestyle='--', alpha=0.7)
plt.tight_layout()
plt.show()

# Plot 2: CAPEX vs Capacity Factor
capex_values = [technologies[tech]['capex'] for tech in tech_names]
cf_values = [technologies[tech]['cf'] for tech in tech_names]

plt.figure(figsize=(8, 5))
plt.scatter(cf_values, capex_values, s=100, color=['blue', 'green', 'red'])
for i, tech in enumerate(tech_names):
    plt.annotate(tech, (cf_values[i], capex_values[i]), xytext=(5, 5), textcoords='offset points', fontsize=10)
plt.title('CAPEX vs Capacity Factor', fontsize=14)
plt.xlabel('Capacity Factor', fontsize=12)
plt.ylabel('CAPEX (USD/kW)', fontsize=12)
plt.grid(True, linestyle='--', alpha=0.7)
plt.tight_layout()
plt.show()

NameError: name 'plt' is not defined